pilot/pkg/model/gateway.go

```diff
@@ -87,10 +90,21 @@ func RecordRejectedConfig(gatewayName string) {
 	totalRejectedConfigs.With(typeTag.Value("gateway"), nameTag.Value(gatewayName)).Increment()
 }
 
+// DisableGatewayPortTranslationLabel is a label on Service that declares that, for that particular
+// service, we should not translate Gateway ports to target ports. For example, if I have a Service
+// on port 80 with target port 8080, with the label. Gateways on port 80 would *not* match. Instead,
+// only Gateways on port 8080 would be used. This prevents ambiguities when there are multiple
+// Services on port 80 referring to different target ports. Long term, this will be replaced by
+// Gateways directly referencing a Service, rather than label selectors. Warning: this label is
+// intended solely for as a workaround for Knative's Istio integration, and not intended for any
+// other usage. It can, and will, be removed immediately after the new direct reference is ready for
+// use.
```

Comment:
How about an annotation or field in the Gateway itself, and keeping it longer term ? 

It may be more readable and safe to be able to look at the Gateway CR and understand that port 80 is external port and 8080 is the local port that is matched against Authz policies. I think that would also address KNative use.

Reply:
Its a bit confusing but it actually needs to apply to the Service I think. The reason is we have these resources:
Service xlb on port 80->8080
Service ilb on port 80->8081
Gateway xlb-gw on port 80
Gateway ilb-gw on port 8081

I think what you are proposing is to annotate ilb-gw, right?

The issue is not in ilb-gw, but xlb-gw. It matches two Services (xlb and ilb) which introduces the ambiguity.

We could annotation xlb-gw I suppose but it seems a bit odd.